### PR TITLE
[MD] func testing for data source management listing

### PR DESCRIPTION
**Pull request by mpabba3003**
[MD] func testing for data source management listing

### Description

Cypress functional tests for datasource management

### Check List

- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by seraphjiang** [comment]
could we create PR against main?

File: cypress/integration/core-opensearch-dashboards/vanilla-opensearch-dashboards/datasource-management-plugin/2_datasource_table.spec.js [hunk 1]
```diff
@@ -0,0 +1,410 @@
+import { MiscUtils } from '@opensearch-dashboards-test/opensearch-dashboards-test-library';
+
+const miscUtils = new MiscUtils(cy);
+
+describe('Datasource Management: Table', () => {
+  beforeEach(() => {
+    // Set welcome screen tracking to false
+    localStorage.setItem('home:welcome:show', 'false');
+
+    // Visit ISM OSD
```
Comment (kristenTian) [inline cypress/integration/core-opensearch-dashboards/vanilla-opensearch-dashboards/datasource-management-plugin/2_datasource_table.spec.js:10]: fix the comments?

File: cypress/integration/core-opensearch-dashboards/vanilla-opensearch-dashboards/datasource-management-plugin/2_datasource_table.spec.js [hunk 2]
```diff
@@ -0,0 +1,410 @@
+import { MiscUtils } from '@opensearch-dashboards-test/opensearch-dashboards-test-library';
+
+const miscUtils = new MiscUtils(cy);
+
+describe('Datasource Management: Table', () => {
+  beforeEach(() => {
+    // Set welcome screen tracking to false
+    localStorage.setItem('home:welcome:show', 'false');
```
Comment (kristenTian) [inline cypress/integration/core-opensearch-dashboards/vanilla-opensearch-dashboards/datasource-management-plugin/2_datasource_table.spec.js:8]: not sure why we need this.

File: cypress/integration/core-opensearch-dashboards/vanilla-opensearch-dashboards/datasource-management-plugin/2_datasource_table.spec.js [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+
+    // Visit ISM OSD
+    miscUtils.visitPage('app/management/opensearch-dashboards/dataSources');
+
+    // Common text to wait for to confirm page loaded, give up to 60 seconds for initial load
+    cy.contains(
+      'Create and manage the data sources that help you retrieve your data from multiple Elasticsearch clusters',
```
Comment (kristenTian) [inline cypress/integration/core-opensearch-dashboards/vanilla-opensearch-dashboards/datasource-management-plugin/2_datasource_table.spec.js:15]: remove `Elasticsearch` references

**Comment by tianleh** [comment]
Please take a look at https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/317 you may need to move the location of the test file and also add a feature flag. See this example https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/cypress/integration/plugins/security/audit_log_spec.js#L11

**Comment by tianleh** [comment]
Please share the Cypress video for this test

**Comment by mpabba3003** [comment]
Closing this PR. I will raise a new PR against Main branch.

**Comment by mpabba3003** [comment]
@seraphjiang @tianleh A new PR #325 has been created which includes resolution for all comments on this PR

**Pull request closed without merging**